src/Rings/mpoly-graded.jl

```diff
@@ -1562,6 +1562,397 @@ end
+
+Create a "homogenizing operator" using the grading specified by the columns of `W`; `h` is the prefix for the homogenizing variables; `pos` indicates where to put the homogenizing variables in the list of generators of the graded polynomial ring (default is after all the other variables).
+
+# Examples
+```jldoctest
+julia> P, (x,y) = polynomial_ring(QQ, ["x", "y"]);
+
+julia> W = ZZMatrix(2,2, [2,3,5,7]);
+
+julia> H = homogenizer(P, W, "h");
+
+julia> H(x^2+y)
+x^2 + y*h[1]*h[2]^3
+
+julia> H([x^2+y])
+1-element Vector{MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}}:
+ x^2 + y*h[1]*h[2]^3
+
+julia> H(ideal([x^2+y]))
+Ideal generated by
+  x^2 + y*h[1]*h[2]^3
+```
+"""
+  function homogenizer(P::MPolyRing{T}, W::Union{ZZMatrix, Matrix{<:IntegerUnion}}, h::VarName; pos::Int=1+ngens(P))  where T
+    n = nvars(P)
+    @req size(W,2) == n  "Weight matrix not compatible with polynomial ring"
+    @req pos in 1:n+1 "Index out of range."
+    vars = copy(symbols(P))
+    grading_dimension = size(W,1)
+    if grading_dimension == 1
+      insert!(vars, pos, Symbol(h))
+    else
+      for i = 1:grading_dimension
+        insert!(vars, pos-1+i, Symbol("$h[$i]"))  # using names like h[1], h[2], etc
+      end
+    end
+    G = abelian_group(zeros(Int, grading_dimension))
+    W = ZZMatrix(W)
+    WH = hcat(Matrix(W[:, 1:(pos-1)]),
+              Matrix(identity_matrix(ZZ, grading_dimension)),
+              Matrix(W[:, pos:n]))
+#    println("weights $([G(WH[:, i]) for i = 1:size(WH, 2)])");
+    P_homog,_ = graded_polynomial_ring(base_ring(P), vars; weights = [G(WH[:, i]) for i = 1:size(WH, 2)])
+
+    VarMap = vcat(1:pos-1, pos+grading_dimension:n+grading_dimension)
+    HVars = collect(pos:pos+grading_dimension-1)
+    return new(P, P_homog, VarMap, HVars, W)
+  end #function
+
+  # PROBABLY NOT USEFUL (create a homogenizer from a dehomogenizer)
+  # function homogenizer(DH::dehomogenizer)
+  #   P = DH.P
+  #   P_homog = DH.P_homog
+  #   VarMap = DH.VarMap
+  #   HVars = DH.HVars;
+  # end
+end # struct
+
+# Returns a list of the homogenizing variables (elements of P_homog)
+function homogenizing_variables(H::homogenizer)
+  HVarList = [gen(H.P_homog,j)  for j in H.HVars]
+  return HVarList;
+end
+
+
+# Make a dehomogenizer from a homogenizer
+# (see comments/doc for struct homogenizer, above)
+struct dehomogenizer
+  P::MPolyRing
+  P_homog::MPolyDecRing
+  VarMap::Vector{Int}
+  HVars::Vector{Int}
+  W::ZZMatrix
+
+  function dehomogenizer(H::homogenizer)
+    return new(H.P, H.P_homog, H.VarMap, H.HVars)
+  end
+end
+
+
+# Use a homogenizer as a function/operator to actually homogenize
+function (H::homogenizer)(f::MPolyRingElem)
+  @req parent(f) == H.P  "Not in correct polynomial ring"
+  return Oscar._homogenization(f, H.P_homog, H.HVars[1])
+end
+
+function (H::homogenizer)(V::Vector{<:MPolyRingElem})
+  # Call below will check arg compatibility: if V is empty, there is nothing to check!
+  return H.(V)
+end
```

Comment:
This does not seem to be super necessary since the user can apply `H.(...)` to begin with. Unless (but the current implementation does not do so at the moment) there is some overhead computation which we can avoid by digesting the whole vector at once.

Reply:
I discussed this briefly yesterday with @wdecker and @jankoboehm .  We concluded that it is nicer to allow also `H(V)` so that Julia newcomers are not obliged to learn the broadcasting syntax (which appears to be a Julia/Matlab foible)

Reply:
I personally think this is not a good idea. We do not do this for anything else and adding this randomly does not help the user in the long run. Yes, it is a julia notation, but we are using julia, don't we?

Reply:
Makes little difference to me.  The Julia/Matlab notation is "cool" but also "alien" to anyone who is new to Julia, so steepens the initial learning curve.  It is also not clear to me how often someone might really want to do `H(V)`.  Of course there is also `[H(f)  for f in V]` which does not explicitly use broadcasting -- I presume the compiler will produce the same code as for broadcasting.  What do @wdecker and @jankoboehm say?

Reply:
What makes little sense to you? That our aim is to build something consistent?

As you said, anyone can write `[H(f) for f in V]`. Or is the argument that this is also too "foreign", because it does not exist in assembly or some other random language?